code/datums/wires/alarm.dm

```diff
@@ -39,74 +26,59 @@
 	. += "The Air Alarm is [(A.shorted || (A.stat & (NOPOWER|BROKEN))) ? "offline." : "working properly!"]"
 	. += "The 'AI control allowed' light is [A.aidisabled ? "off" : "on"]."
 
-/datum/wires/alarm/UpdateCut(index, mended)
+/datum/wires/alarm/on_cut(wire, mend)
 	var/obj/machinery/alarm/A = holder
-	switch(index)
-		if(AALARM_WIRE_IDSCAN)
-			if(!mended)
+	switch(wire)
+		if(WIRE_IDSCAN)
+			if(!mend)
 				A.locked = 1
-//				to_chat(world, "Idscan wire cut")
 
-		if(AALARM_WIRE_POWER)
+		if(WIRE_MAIN_POWER1)
 			A.shock(usr, 50)
-			A.shorted = !mended
+			A.shorted = !mend
 			A.update_icon()
-//			to_chat(world, "Power wire cut")
 
-		if(AALARM_WIRE_AI_CONTROL)
-			A.aidisabled = !mended
-//				to_chat(world, "AI Control Wire Cut")
+		if(WIRE_AI_CONTROL)
+			A.aidisabled = !mend
 
-		if(AALARM_WIRE_SYPHON)
-			if(!mended)
+		if(WIRE_SYPHON)
+			if(!mend)
 				A.mode = 3 // AALARM_MODE_PANIC
 				A.apply_mode()
-//				to_chat(world, "Syphon Wire Cut")
 
-		if(AALARM_WIRE_AALARM)
+		if(WIRE_AALARM)
 			if(A.alarm_area.atmosalert(ATMOS_ALARM_DANGER, A))
 				A.post_alert(ATMOS_ALARM_DANGER)
 			A.update_icon()
 	..()
 
-/datum/wires/alarm/UpdatePulsed(index)
+/datum/wires/alarm/on_pulse(wire)
 	var/obj/machinery/alarm/A = holder
-	switch(index)
-		if(AALARM_WIRE_IDSCAN)
+	switch(wire)
+		if(WIRE_IDSCAN)
 			A.locked = !A.locked
-//			to_chat(world, "Idscan wire pulsed")
 
-		if(AALARM_WIRE_POWER)
-//			to_chat(world, "Power wire pulsed")
-			if(A.shorted == 0)
-				A.shorted = 1
+		if(WIRE_MAIN_POWER1)
+			if(!A.shorted)
+				A.shorted = TRUE
 				A.update_icon()
+			addtimer(CALLBACK(A, /obj/machinery/alarm/.proc/unshort_callback), 120 SECONDS)
```

Comment:
You used `120 SECONDS` here, but the old value was `12000`, which is 12000 seconds. Though saying that, an air alarm being shorted for **20 minutes** seems off, so I prefer this value.

Reply:
This was spoken about on discord, and is now declared a fix since its the same timer for APCs.